test/integration/puppetlabs/puppetserver/testutils.clj

```diff
@@ -207,21 +210,41 @@
   (= name (get catalog "name")))
 
 (schema/defn ^:always-validate resource-matches? :- schema/Bool
-  [resource-type :- schema/Str
-   resource-title :- schema/Str
-   resource :- PuppetResource]
-  (and (= resource-type (resource "type"))
-       (= resource-title (resource "title"))))
+  ([resource-type :- schema/Str
+    resource-title :- schema/Str
+    resource :- PuppetResource]
+   (and (= resource-type (resource "type"))
+        (= resource-title (resource "title"))))
+  ([resource-type :- schema/Str
+    resource-title :- schema/Str
+    resource :- PuppetResource
+    param-name :- schema/Str
+    param-value :- schema/Any]
+   (and
+    (resource-matches? resource-type resource-title resource)
+    (= param-value (get-in resource ["parameters" param-name])))))
 
 (schema/defn ^:always-validate catalog-contains? :- schema/Bool
-  [catalog :- PuppetCatalog
-   resource-type :- schema/Str
-   resource-title :- schema/Str]
-  (let [resources (get catalog "resources")]
-    (->> resources
-        (some (partial resource-matches? resource-type resource-title))
-        nil?
-        not)))
+  ([catalog :- PuppetCatalog
+    resource-type :- schema/Str
+    resource-title :- schema/Str]
+   (let [resources (get catalog "resources")]
+     (->> resources
+          (some (partial resource-matches? resource-type resource-title))
+          nil?
+          not)))
+  ([catalog :- PuppetCatalog
+    resource-type :- schema/Str
+    resource-title :- schema/Str
+    param-name :- schema/Str
+    param-value :- schema/Any]
+   (let [resources (get catalog "resources")]
+     (some #(resource-matches? resource-type resource-title % param-name param-value) resources))))
+
+(schema/defn catalog-code-id :- (schema/maybe schema/Str)
```

Comment:
It isn't currently used, but it seemed to me like keeping all of the catalog helpers in one repo had benefits. I could also foresee tests being added for direct puppet work that exercised the helper.